### PR TITLE
Implement organization-wide issue labeling scheme

### LINKED ISSUE #1: Proposed Issue Label Scheme
In order to better triage issues and provide some guidance for would-be contributors, I'd like to propose that we adopt an issue labeling scheme like the ones used by the [Rust project](https://github.com/rust-lang/rust/labels), the [Habitat project](https://github.com/habitat-sh/habitat/labels), and the [Kubernetes project](https://github.com/kubernetes/kubernetes/labels). 

Reasoning for this scheme (for the Habitat project, at least) can be found [here](https://www.habitat.sh/blog/2017/06/Triage-and-Contributions/) and [here](https://www.habitat.sh/blog/2017/06/Habitat-Issue-Tracker/)

For Habitat, we use [repo_man](https://github.com/thommay/repo_man) to manage the labels in an automated way. One nice thing about this tool is that it operates across multiple repositories, meaning that we could maintain a consistent set of labels across all the Cog-related repositories. Additionally, it is driven from a configuration file, meaning that we can manage our scheme as code, perhaps in this very repository.

Below is a list of proposed labels that we could use. Please suggest additional labels as you see fit:

- area/executor
- area/adaptor
- area/adaptor/slack
- area/adaptor/hipchat
- area/api
- area/commands
- area/rules
- area/relay
- area/build (e.g. CI, release tooling, etc)
- area/testing (things to help command authors test easily)
- area/documentation
- effort/easy
- effort/medium
- effort/difficult
- type/bug
- type/chore
- type/feature
- language/elixir
- language/go
- language/ruby 
- language/python

In addition to the "scoped" labels above, there are probably a few general labels we could use, as well:

- help-wanted
- needs-discussion
- waiting-for-response

If we adopt this scheme, we should also document the intention and meaning of each label.

### PR DESCRIPTION
**Pull request by christophermaier**
Fixes #1

### CODE REVIEW

File: labels/README.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  Issues specific to the Slack adapter
+* `area/api`
+  The Cog REST API
+* `area/build`
+  Issues around CI, release tooling, packaging, etc.
+* `area/cli`
+  The `cogctl` command-line interface
+* `area/commands`
+  Chat commands
+* `area/documenataion`
```
Comment (ohaiwalt) [inline labels/README.md]: `s/documenataion/documentation/`

**Review by ohaiwalt — DISMISSED** [review]
Looks great pending one spelling fix!

**Comment by christophermaier** [comment]
@ohaiwalt Thanks for catching that spelling error. I've fixed that and am going to merge.

**Comment by jesselang** [comment]
Outstanding, thanks all!

**Pull request merged**